File: src/main/javascript/processing/bugfest/session/session-2/Nadine.js
// function setup() {
// 	createCanvas(windowWidth, windowHeight);
// 	var r = random(255);
//   var g = random(255);
//   var b = random(255);
// 	background(r, g, b);
// 	// line(100,100,500,500);
// 	// rect(0, 0, 500, 500);
// 	// translate(300, 200);
// 	// rect(0, 0, 400, 400);
// 	// translate(250, 150);
// 	// rect(0, 0, 300, 300);
// 	// for (let i = 0; i < 3; i++)
// 	// {
// 	// 	square(100, 100, 100*i, 100*i);
// 	// }
// 	// for (let i = 0; i < 9; i++)
// 	// {
// 	// 	rect(0, 0, random(500), random(500));
// 	// 	translate(random(250), random(250))
// 	// }

// 	// bug creation
// 	let p1 = { x: random(300), y: random(300) };
// 	let p2 = { x: random(500), y: random(500) };
// 	let p3 = { x: random(700), y: random(700) };
// 	noFill();
// 	stroke(random(255), random(255), random(255));
// 	strokeWeight(random(10));
// 	curve(p1.x, p1.y, p1.x, p1.y, p2.x, p2.y, p3.x, p3.y);
// }

// function draw() {
// 	circle(mouseX, mouseY, 20);
// }

let beginX = 20.0; // Initial x-coordinate
let beginY = 10.0; // Initial y-coordinate
let endX = 570.0; // Final x-coordinate
let endY = 320.0; // Final y-coordinate
let distX; // X-axis distance to move
let distY; // Y-axis distance to move
let x = 0.0; // Current x-coordinate
let y = 0.0; // Current y-coordinate
let step = 0.005; // Size of each step along the path
let pct = 0.0; // Percentage traveled (0.0 to 1.0)

function setup() {
  createCanvas(720, 400);
	var r = random(255);
  var g = random(255);
  var b = random(255);
	background(r, g, b);
  noStroke();
  distX = endX - beginX;
  distY = endY - beginY;
}

function draw() {
  fill(0, 2);
  pct += step;
	exponent = random(20);
  if (pct < 1.0) {
    x = beginX + pct * distX;
    y = beginY + pow(pct, exponent) * distY;
  }
  fill(255);
	radius = 20
	point_size = random(20)/20
  ellipse(x, y, radius*point_size, radius*point_size);
}

File: src/main/java/bugfest/s1768386/Nadine.java
package bugfest.s1768386;

import bugfest.utilities.PdfRecorder;
import processing.core.PApplet;

// https://openprocessing.org/sketch/1768386
public class Nadine extends PdfRecorder {
    float beginX = 20.0f; // Initial x-coordinate
    float beginY = 10.0f; // Initial y-coordinate
    float endX = 570.0f; // Final x-coordinate
    float endY = 320.0f; // Final y-coordinate
    float distX; // X-axis distance to move
    float distY; // Y-axis distance to move
    float x = 0.0f; // Current x-coordinate
    float y = 0.0f; // Current y-coordinate
    float step = 0.005f; // Size of each step along the path
    float pct = 0.0f; // Percentage traveled (0.0 to 1.0)

    public Nadine() {
        super(Nadine.class.getName());
    }

    public static void main(String[] arguments) {
        PApplet.main(Nadine.class);
    }

    @Override
    public void settings() {
        size(720, 400);
    }

    @Override
    public void setup() {
        beginRecordPdf();

        var r = random(255);
        var g = random(255);
        var b = random(255);
        background(r, g, b);
        noStroke();
        distX = endX - beginX;
        distY = endY - beginY;
    }

    @Override
    public void draw() {
        fill(0, 2);
        pct += step;
        float exponent = random(20);
        if (pct < 1.0) {
            x = beginX + pct * distX;
            y = beginY + pow(pct, exponent) * distY;
        }
        fill(255);
        float radius = 20;
        float point_size = random(20) / 20;
        ellipse(x, y, radius * point_size, radius * point_size);
    }

    @Override
    public void keyPressed() {
        if (key == 'p') {
            endRecordPdf();
            beginRecordPdf();
        }
    }
}
